### ISSUE TITLE
losses.binary_focal_loss_with_logits is computed elementwise

### ISSUE THREAD

**Issue by jatentaki** [kornia/kornia#1308]
### Describe the bug

Due to an apparent bug involving `unsqueeze`, batched `binary_focal_loss_with_logits` is computed pairwise along the batch dimension.

### Reproduction steps

```bash
I created a colab illustrating the issue. Please note how removing the two (un)squeezing lines seems to solve the issue. I can make a PR if wanted but I'm asking beforehand because I don't understand the rationale that got those unsqueezing operations to appear in the first place.

https://colab.research.google.com/drive/19OpQxpsJJfki7hV2Ht-3vDM6RO9c8x2W?usp=sharing
```

### Expected behavior

Loss for inputs of shape `(B, 1, ...)` should be of shape `(B, 1, ...)` or `(B, ...)` but not `(B, B, 1, ...)`.

### Environment

```shell
PyTorch version: 1.10.0.dev20210921+cu111
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.8.10 (default, Jun  4 2021, 15:09:15)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.11.0-1018-gcp-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: Tesla V100-SXM2-16GB
GPU 1: Tesla V100-SXM2-16GB
GPU 2: Tesla V100-SXM2-16GB
GPU 3: Tesla V100-SXM2-16GB
GPU 4: Tesla V100-SXM2-16GB
GPU 5: Tesla V100-SXM2-16GB
GPU 6: Tesla V100-SXM2-16GB
GPU 7: Tesla V100-SXM2-16GB

Nvidia driver version: 470.57.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.20.3
[pip3] pytorch-lightning==1.4.7
[pip3] torch==1.10.0.dev20210921+cu111
[pip3] torch-dimcheck==0.0.1
[pip3] torchmetrics==0.4.1
[pip3] torchvision==0.11.0.dev20210921+cu111
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.1.74              h6bb024c_0    nvidia
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2021.3.0           h06a4308_520  
[conda] mkl-service               2.4.0            py38h7f8727e_0  
[conda] mkl_fft                   1.3.0            py38h42c9631_2  
[conda] mkl_random                1.2.2            py38h51133e4_0  
[conda] mypy-extensions           0.4.3                    pypi_0    pypi
[conda] numpy                     1.19.5                   pypi_0    pypi
[conda] numpy-base                1.20.3           py38h74d4b33_0  
[conda] pytorch-lightning         1.4.7                    pypi_0    pypi
[conda] torch                     1.10.0.dev20210921+cu111          pypi_0    pypi
[conda] torchmetrics              0.4.1                    pypi_0    pypi
[conda] torchvision               0.11.0.dev20210921+cu111          pypi_0    pypi
```

**Comment by edgarriba** [comment]
@jatentaki mind sending a PR with the fix. Besides, I''ll try to test tomorrow with a full end to end example for segmentation with the new training api :)

See: https://github.com/kornia/kornia/pull/1307
